### PR TITLE
Adds Features Search Parts and Open filename in title bar, and fixes bug with closing OOXML Parts

### PR DESCRIPTION
**Pull request by mikeebowen**
This pull request:
- Adds feature #17 
- Adds feature #22
- Fixes bug #24 
  - The fix for this is a hack to get around 2 known bugs/issues in VS Code. If they're fixed this should be reverted. Details in comments in the code.

### CODE REVIEW

**Comment by mikeebowen** [comment]
Apologies for the console log commits. The tests passed locally, but were failing on the Azure pipeline, so it was the only way to get any info from the failing tests. It was an issue with creating too many instances of the viewer for testing, so I moved the viewer instance creator to suiteSetup and that solves the issue.

**Review by yuenm18 — COMMENTED** [review]
Thanks for the PR. It looks mostly good. I just have a few suggestions on naming and syntax

File: src/ooxml-viewer.ts [hunk 1]
```diff
@@ -462,8 +492,47 @@ export class OOXMLViewer {
...
+  /**
+   * @description search OXML parts for a string and display the results in a web view
+   * @method searchOxmlParts
+   * @returns {Promise<void>}
+   */
+  async searchOxmlParts(): Promise<void> {
+    const warningMsg = 'A file must be open in the OOXML Viewer to search its parts.';
+    try {
+      await workspace.fs.stat(Uri.file(this.cache.normalSubfolderPath));
+      const searchTerm = await window.showInputBox({ title: 'Search OXML Parts', prompt: 'Enter a search term.' });
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-viewer.ts:510]: Fixed, typo...

File: src/ooxml-file-cache.ts [hunk 2]
```diff
@@ -280,12 +284,18 @@ export class OOXMLFileCache {
...
    * @returns {Promise<void>}
    */
-  async writeFile(cachedFilePath: string, fileContents: Uint8Array): Promise<void> {
+  async writeFile(cachedFilePath: string, fileContents: Uint8Array, throwError = false): Promise<void> {
     try {
       await workspace.fs.createDirectory(Uri.file(dirname(cachedFilePath)));
       await workspace.fs.writeFile(Uri.file(cachedFilePath), fileContents);
     } catch (err) {
-      console.error(`Unable to create file '${cachedFilePath}'`, err);
+      if (throwError) {
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-file-cache.ts:292]: I like handling the error in only one place, but this syntax isn't quite valid. `throwError && throw new Error(err);` the throw after the and operator causes an "Expression expected." error from TypeScript and [throwing an exception immediately hands control back to the caller](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw), so if `throwError` is true then nothing after the error is thrown will be called. I refactored it so the syntax is valid.

```javascript
if (!throwError) {
  const msg = `Unable to create file '${cachedFilePath}\n ${err.message || err}`;
  console.error(msg);
  await window.showErrorMessage(msg);
} else {
  throw new Error(err);
}
```

File: src/ooxml-file-cache.ts [hunk 3]
```diff
@@ -28,6 +28,10 @@ export class OOXMLFileCache {
     return join(this.context.storageUri?.fsPath || '', CACHE_FOLDER_NAME);
   }
 
+  get normalSubfolderPath(): string {
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-file-cache.ts:31]: Done

File: src/ooxml-file-cache.ts [hunk 4]
```diff
@@ -299,7 +309,9 @@ export class OOXMLFileCache {
     try {
       await workspace.fs.delete(Uri.file(cachedFilePath), { recursive: true, useTrash: false });
     } catch (err) {
-      console.error(`Unable to delete file '${cachedFilePath}'`, err);
+      const msg = `Unable to delete file '${cachedFilePath}\n ${err.message || err}`;
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-file-cache.ts:312]: Got rid of the extra space at the start of the lines.

File: src/ooxml-viewer.ts [hunk 5]
```diff
@@ -462,8 +492,47 @@ export class OOXMLViewer {
...
 
     return false;
   }
+
+  /**
+   * @description search OXML parts for a string and display the results in a web view
+   * @method searchOxmlParts
+   * @returns {Promise<void>}
+   */
+  async searchOxmlParts(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-viewer.ts:506]: Changed

File: src/ooxml-file-cache.ts [hunk 6]
```diff
@@ -280,12 +284,18 @@ export class OOXMLFileCache {
...
+  async writeFile(cachedFilePath: string, fileContents: Uint8Array, throwError = false): Promise<void> {
     try {
       await workspace.fs.createDirectory(Uri.file(dirname(cachedFilePath)));
       await workspace.fs.writeFile(Uri.file(cachedFilePath), fileContents);
     } catch (err) {
-      console.error(`Unable to create file '${cachedFilePath}'`, err);
+      if (throwError) {
+        throw new Error(err);
+      } else {
+        const msg = `Unable to create file '${cachedFilePath}\n ${err.message || err}`;
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-file-cache.ts:295]: Spaced removed.

File: src/ooxml-file-cache.ts [hunk 7]
```diff
@@ -313,7 +325,9 @@ export class OOXMLFileCache {
     try {
       return await workspace.fs.readFile(Uri.file(cachedFilePath));
     } catch (err) {
-      console.error(`Unable to read file '${cachedFilePath}'`, err);
+      const msg = `Unable to read file '${cachedFilePath}\n ${err.message || err}`;
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-file-cache.ts:328]: Space removed.

File: src/ooxml-viewer.ts [hunk 8]
```diff
@@ -462,8 +492,47 @@ export class OOXMLViewer {
...
     } catch (err) {
       console.error(err.message || err);
+      await window.showErrorMessage(err.message || err);
     }
 
     return false;
   }
+
+  /**
+   * @description search OXML parts for a string and display the results in a web view
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-viewer.ts:502]: Fixed

File: src/ooxml-viewer.ts [hunk 9]
```diff
@@ -462,8 +492,47 @@ export class OOXMLViewer {
...
+   * @method searchOxmlParts
```
[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-viewer.ts:503]: Changed.

File: src/ooxml-file-cache.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (yuenm18) [reply src/ooxml-file-cache.ts]: can you add a single quote after `${cachedFilePath}`

[the remark this answers is not in the document]
Comment (mikeebowen) [reply src/ooxml-file-cache.ts]: Added 187963aa2a7ba56fcaccc3ca125c317a05b0fe8c

**Pull request merged**